### PR TITLE
Lb je issue nine

### PR DESCRIPTION
**Pull request by Lizhtml**
A filter is created for products from the shopping list. 
the following requirements are taken into account 

AC:

Display a text field above the top of the shopping list
As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
The filter text should match any part of the item name (i.e. it should not only match from the start of the string)

**For this a library is used.**

image

### CODE REVIEW

File: src/components/Product/Product.js
Comment (DNeosphere) [inline src/components/Product/Product.js]: I didn't know about this module, looks interesting

File: src/components/Product/Product.js
Comment (DNeosphere) [inline src/components/Product/Product.js:27]: Some debugging happening here, do you guys need this line in production? :trollface:

**Comment by DNeosphere** [comment]
I like the fact that you took time to search for solutions and found a module that provides an autocomplete component, however, the process to accomplish that component from scratch, could improve your problem solution skills.

File: src/components/Product/Product.js
Comment (Lizhtml) [reply src/components/Product/Product.js:27]: jajajajajajaaj ups!!  Thank you for notifying us about this.

File: src/components/Product/Product.js
Comment (arche4) [inline src/components/Product/Product.js:81]: Cool that they organized this. 💛

**Review by alejandronanez — CHANGES_REQUESTED** [review]
👋 @Lizhtml @AndrewEspejo 

We need to make some changes before we merge this! I've detailed what needs to be done in this video. https://www.loom.com/share/1c0c656c1ce34533add1ea1f5bc5d1f7

Please, let me know if you have any question!

@AndrewEspejo I apologize, in the video I said "Liz + Yesika" but I should had said "Liz + Andrés" 😓. Sorry about that!

**Comment by Lizhtml** [comment]
@alejandronanez  Thanks Alejandro, we will review it to upload it as soon as possible.

File: src/components/Product/ListProduct.js
Comment (alejandronanez) [inline src/components/Product/ListProduct.js:18]: Is there any reason to use `var` and not `let` here?

File: src/components/Product/ListProduct.js
Comment (alejandronanez) [inline src/components/Product/ListProduct.js:25]: Question here, any reason tu use `function` and not `() => {....}`

File: src/components/Product/Product.js
Comment (alejandronanez) [reply src/components/Product/Product.js:27]: OMG I didn't know :trollface: was a thing! LOL

But seriously, you need to delete that `console.log` :P

File: src/components/Product/Product.js
Comment (alejandronanez) [reply src/components/Product/Product.js:81]: For future occasions: 

Consider giving Regular Expressions a meaningful name, for example:

`const LOWERCASE_NAMES_ONLY = /..../`;

**Review by alejandronanez — APPROVED** [review]
Nice work y'all!

I left a couple of questions about some decisions you took, but they should not block the merge of this PR.

**Comment by Lizhtml** [comment]
closes by mistake

**Pull request merged**
